src/Microsoft.Diagnostics.EventFlow.Core/Implementations/EventFlowSubject.cs

```diff
@@ -47,24 +64,24 @@ public void OnError(Exception error)
 
         public void OnNext(SubjectType value)
```

Comment:
```OnNext``` can be run in multiple threads right? In this case, I can see that the state of ```notifyingObservers``` is just not correctly maintained in the original implementation, even without the ```ShutDown()``` method. E.g., thread 1 at line 57, thread 2 at line 67, ```notifyingObservers``` is set to false but it's actually still notfiying the observers.
The new implementation looks much better.

Reply:
Yes, that is right, good point!